src/sources/kubernetes_logs/mod.rs

```diff
@@ -191,20 +191,56 @@ impl Source {
         let annotator = PodMetadataAnnotator::new(state_reader, fields_spec);
 
         // TODO: maybe some of the parameters have to be configurable.
+
+        // The 16KB is the maximum size of the payload at single line for both
+        // docker and CRI log formats.
+        // We take a double of that to account for metadata and padding, and to
+        // have a power of two rounding. Line splitting is countered at the
+        // parsers, see the `partial_events_merger` logic.
         let max_line_bytes = 32 * 1024; // 32 KiB
         let file_server = FileServer {
+            // Use our special paths provider.
             paths_provider,
+            // This is the default value for the read buffer size.
             max_read_bytes: 2048,
-            start_at_beginning: true,
+            // We want to use checkpoining mechanism, and resume from where we
+            // left off.
+            start_at_beginning: false,
```

Comment:
Why are we not relying on the `Default` implementation here? That would have avoided this bug in the first place.

Reply:
`FileServer` does not implement `Default`.
I know you've asked this before, and I was confused by this question last time, so might have forgot to address it, sorry.

I might still be confused, in that case could you please elaborate?

Reply:
Indeed, `FileServer` does not have a safe constructor with defaults.

I think adding such would be a separate ticket, which I feel should be done, since right now it's very difficult to use the `FileServer` type in a safe way.

Reply:
> I think adding such would be a separate ticket, which I feel should be done, since right now it's very difficult to use the `FileServer` type in a 'default' way.

I'm not sure if default for the whole thing makes sense. But we could extract a subset of parameters into a `Default`-constructible object and pass that.

However, even then, for something like a `kubernetes_logs` source, I'd assign most of the settings manually. For instance, this one should probably be set in the `source` level rather than being defined at the `Default` that may change independently of the source (and potentially break the compatibility).

Some of the options I'd definitely offload to `Default`, like `max_read_bytes`.

We also need a proper test for this behaviour at the E2E, I'll create an issue.

Reply:
Sorry, I should have been clearer. We have encoded a reasonable set of default values into the `Default` implementation for `FileConfig`. It would take some tweaking, but we should really try to rely on those defaults wherever the source is reused.

Reply:
Ah, makes sense! `FileConfig` isn't exactly a good fit for the use in `kubernetes_logs` source, so we probably need to add a layer of indirection.
I'll create an issue for it.